roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -95,29 +95,30 @@ impl JDsMempool {
     }
 
     pub async fn update_mempool(self_: Arc<Mutex<Self>>) -> Result<(), JdsMempoolError> {
-        let mut mempool_ordered: HashMap<Txid, Option<Transaction>> = HashMap::new();
+        let mut new_jds_mempool: HashMap<Txid, Option<Transaction>> =
+            self_.safe_lock(|x| x.mempool.clone())?;
+
+        // the fat transactions in the jds-mempool are those declared by some downstream and we
+        // don't want to remove them, but we can get rid of transactions that are none
+        new_jds_mempool.retain(|_, val| val.is_some());
 
         let client = self_
             .safe_lock(|x| x.get_client())?
             .ok_or(JdsMempoolError::NoClient)?;
 
-        let mempool: Vec<String> = client.get_raw_mempool().await?;
-        for id in &mempool {
+        let node_mempool: Vec<String> = client.get_raw_mempool().await?;
+
+        // Add new transactions from the node's mempool
+        for id in &node_mempool {
             let key_id = Txid::from_str(id)
                 .map_err(|err| JdsMempoolError::Rpc(RpcError::Deserialization(err.to_string())))?;
-
-            let tx = self_.safe_lock(|x| match x.mempool.get(&key_id) {
-                Some(entry) => entry.clone(),
-                None => None,
-            })?;
-
-            mempool_ordered.insert(key_id, tx);
+            new_jds_mempool.entry(key_id).or_insert(None);
```

Comment:
Where is the tx data inserted?
Maybe I'm missing something

Reply:
Oh I am going to make modifications to it now, I don't know how I missed it. The idea was not adding the data if the key_id exist, but I see that was not what was pushed. I apologise